shotover-proxy/src/tls.rs

```diff
@@ -67,12 +67,16 @@ impl TlsConnector {
         })
     }
 
-    pub async fn connect(&self, tcp_stream: TcpStream) -> Result<SslStream<TcpStream>> {
+    pub async fn connect(
+        &self,
+        tcp_stream: TcpStream,
+        verify_hostname: bool,
```

Comment:
Can we make setting this to false scarier?
Maybe a separate scary sounding method name?
Maybe a seperate constructor for TLSAcceptor?
Or maybe replace the bool with an enum with a scary sounding variant?

As is, its quite easy to accidentally drop an important security measure by copy-pasting the connect call from redis which doesnt give any context to the false value it uses.